app/src/main/java/de/othr/plantico/ui/ownedPlantList/OwnedPlantListAdapter.kt

```diff
@@ -0,0 +1,163 @@
+import java.util.*
+
+
+class OwnedPlantListAdapter(context: Context) :
+    ListAdapter<OwnedPlant, OwnedPlantListAdapter.OwnedPlantViewHolder>(OwnedPlantsComparator()) {
+    private val con: Context = context
+    private var plantList: List<Plant>? = null
+
+    fun addPlants(plants: List<Plant>) {
+        plantList = plants
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OwnedPlantViewHolder {
+        val binding = ViewOwnedPlantItemBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+
+        return OwnedPlantViewHolder(binding, con)
+    }
+
+    override fun onBindViewHolder(holder: OwnedPlantViewHolder, position: Int) {
+        val current = getItem(position)
+        holder.bind(current, position)
+    }
+
+
+    inner class OwnedPlantViewHolder(binding: ViewOwnedPlantItemBinding, context: Context) :
+        RecyclerView.ViewHolder(
+            binding.root
+        ),
+        View.OnClickListener {
+
+        private val itemBinding = binding
+
+        @SuppressLint("SetTextI18n")
+        fun bind(plant: OwnedPlant, position: Int) {
+            itemBinding.plantTitleText.text = plant.plantName
+            itemBinding.Room.text = plant.location ?: "-"
+            if (position == 0 || getItem(position - 1).location?.lowercase() == plant.location?.lowercase()) {
+                itemBinding.Room.visibility = View.GONE
+            }
+            if (plantList != null) {
+                val realPlant = plantList!![plant.plantID]
+                //Watering
+                val waterdropplettFull = ContextCompat
+                    .getDrawable(con, R.drawable.ic_waterdropplett_full_icon)
+                val waterdropplettEmpty = ContextCompat
+                    .getDrawable(con, R.drawable.ic_waterdropplett_empty_icon)
+
+                when (realPlant.wateringLevel) {
+
+                    WateringLevel.LOW -> {
+                        itemBinding.wateringLight.setImageDrawable(waterdropplettFull)
+                        itemBinding.wateringMedium.setImageDrawable(waterdropplettEmpty)
+                        itemBinding.wateringHeavy.setImageDrawable(waterdropplettEmpty)
+                    }
+                    WateringLevel.MEDIUM -> {
+                        itemBinding.wateringLight.setImageDrawable(waterdropplettFull)
+                        itemBinding.wateringMedium.setImageDrawable(waterdropplettFull)
+                        itemBinding.wateringHeavy.setImageDrawable(waterdropplettEmpty)
+                    }
+                    WateringLevel.HIGH -> {
+                        itemBinding.wateringLight.setImageDrawable(waterdropplettFull)
+                        itemBinding.wateringMedium.setImageDrawable(waterdropplettFull)
+                        itemBinding.wateringHeavy.setImageDrawable(waterdropplettFull)
+                    }
+                }
```

Comment:
We also might consider to outsource this at some point, since it is being used in multiple ListsAdapters.